nconnect.go

```diff
@@ -562,5 +596,129 @@ func (nc *nconnect) SetTunaNode(node *types.Node) {
 }
 
 func (nc *nconnect) GetTunnels() []*tunnel.Tunnel {
-	return nc.tunnels
+	return append(nc.tunnels, nc.networkTunnels...)
+}
+
+func (nc *nconnect) StartNetworkManager() error {
+	m, err := network.NewManager(nc.opts)
+	if err != nil {
+		return err
+	}
+
+	go m.StartManager()
+
+	err = m.StartWebServer()
+	if err != nil {
+		return err
+	}
+
+	nc.waitForSignal()
+
+	return nil
+}
+
+func (nc *nconnect) StartNetworkMember() error {
+	if nc.opts.ManagerAddress == "" {
+		return errors.New("network manager address is not specified")
+	}
+
+	mc, err := nc.getAdminClient()
+	if err != nil {
+		return err
+	}
+
+	networkMember, err := network.NewMember(nc.opts, mc) // network member
+	if err != nil {
+		return err
+	}
+
+	nc.networkMember = networkMember
+
+	if err = networkMember.JoinNetwork(); err != nil {
+		return err
+	}
+
+	if err = nc.SetupNetworkTunnel(); err != nil {
+		return err
+	}
+	networkMember.CbNodeICanAccessUpdated = nc.SetupNetworkTunnel
+
+	if err = networkMember.StartMember(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (nc *nconnect) SetupNetworkTunnel() error {
+	addresses, err := nc.networkMember.GetAddressICanAccess()
+	if err != nil {
+		return err
+	}
+	if len(addresses) == 0 {
+		return nil
+	}
+
+	existTunnels := make(map[string]*tunnel.Tunnel)
+	for _, t := range nc.networkTunnels {
+		existTunnels[t.ToAddr()] = t
+	}
+
+	if nc.opts.Verbose {
+		log.Printf("nconnect SetupNetworkTunnels to addresses: %v\n", addresses)
+	}
+
+	var from, to []string
+	for _, addr := range addresses {
+		if _, ok := existTunnels[addr]; ok {
+			delete(existTunnels, addr)
+			continue
+		}
+
+		port, err := util.GetFreePort(0)
+		if err != nil {
+			return err
+		}
+		ssAddr := "127.0.0.1:" + strconv.Itoa(port)
+
+		toAddr := strings.Replace(addr, "client.", "", -1) // remove network member's client. prefix, it should be server's address
+		adminAddr := nc.opts.AdminIdentifier + "." + toAddr
+		if remoteInfo, err := nc.getRemoteInfo(adminAddr); err == nil {
```

Comment:
For network mode, why we need to use other members' local IP at all? We should use the assigned IP address, which doesn't need getRemoteInfo at all

Reply:
Yes, modified.